src/main/webapp/app/exercises/text/assess/analytics/text-assesment-analytics.service.ts

```diff
@@ -0,0 +1,82 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { TextAssessmentService } from 'app/exercises/text/assess/text-assessment.service';
+import { TextAssessmentEvent, TextAssessmentEventType } from 'app/entities/text-assesment-event.model';
+import { AccountService } from 'app/core/auth/account.service';
+import { FeedbackType } from 'app/entities/feedback.model';
+import { TextBlockType } from 'app/entities/text-block.model';
+import { tap, filter } from 'rxjs/operators';
+import { ProfileInfo } from 'app/shared/layouts/profiles/profile-info.model';
+import { ProfileService } from 'app/shared/layouts/profiles/profile.service';
+
+/**
+ * A service used to manage sending TextAssessmentEvent's to the server
+ */
+@Injectable({ providedIn: 'root' })
+export class TextAssessmentAnalytics {
+    private userId: number;
+    private courseId: number;
+    private textExerciseId: number;
+    private participationId: number;
+    private submissionId: number;
+    private eventToSend: TextAssessmentEvent = new TextAssessmentEvent();
+    private INVALID_VALUE = -1;
+    private route: ActivatedRoute;
+    public analyticsEnabled: boolean;
+
+    constructor(protected assessmentsService: TextAssessmentService, protected accountService: AccountService, private profileService: ProfileService) {
+        // retrieve the analytics enabled status from the profile info and set to current property
+        this.profileService
+            .getProfileInfo()
+            .pipe(
+                filter(Boolean),
+                tap((info: ProfileInfo) => {
+                    this.analyticsEnabled = Boolean(info.textAssessmentAnalyticsEnabled);
```

Comment:
What happens if this value is not configured? What is the default?

Reply:
I have set the default to false